client/index.js

```diff
@@ -1,12 +1,14 @@
 var url = require('url');
 var io = require("socket.io-client");
 var stripAnsi = require('strip-ansi');
-var scriptElements = document.getElementsByTagName("script");
 
-var urlParts = url.parse(typeof __resourceQuery === "string" && __resourceQuery ?
-	__resourceQuery.substr(1) :
-	scriptElements[scriptElements.length-1].getAttribute("src").replace(/\/[^\/]+$/, "")
-);
+var urlParts;
+if (typeof __resourceQuery === "string" && __resourceQuery) {
+	urlParts = url.parse(__resourceQuery.substr(1));
+} else {
+	var scriptElements = document.getElementsByTagName("script");
```

Comment:
Getting `<script>` elements is now behind a conditional to prevent web workers from accessing `document` (which would throw an error)